### DISCUSSION TITLE
SciChart vs Oxyplot

### DISCUSSION THREAD

**Discussion by rnugent3** [HydrologicEngineeringCenter/HEC-FDA#273] [Q&A]
Hi All, 

We need to make a decision about which to use in the FDA user interface: SciChart or Oxyplot. Please describe the pros and cons as you see them. 

One caveat: Cody would like to keep SciChart in the impact area scenario editor as that plotting function works well. 

Thanks @Brennan1994 @dezidizon @HenryGeorgist @codymccoy @RyanM-RMA

**Comment by HenryGeorgist** [comment]
I would say that what we have with oxyplot for the editors is working well also and we probably shouldn't support both. I don't know what the work involved in updating table with plot would be to connect scicharts instead of oxyplot. I seem to remember @codymccoy  showing basically one or two methods that seemed to be necessary to implement.

Do we have a set of criteria to define the needs that should be met by a plotting library?

For example:

1. must display x and y data with uncertainty bounds
2. must have a legend
3. must have invertible x axis
3. others?

**Reply by rnugent3** [reply]
I think the required criteria are enumerated in the design document, things like axis labels, be able to copy a plot (the data or an image). I think one thing missing from the design document is that the information in the plot should be equally accessible. 

There are contract implications that we can’t ignore if we don’t want to support both. If you’re willing to do the re-work in the IAS editor, I think we’re good.

**Reply by HenryGeorgist** [reply]
I am not ready to take on replacing sci charts with oxyplot or oxyplot with scicharts. I would like to evaluate the pros and cons but I wanted a bit more structure to help evaluate that. I think your suggestion that if we implement sci charts in table with plot or oxyplot in the IAS control it would have contract repercussions is an important criteria to consider.
I think some additional criteria would be continuity of user experience, ease of continued maintenance, and meets criteria enumerated in the design document.

**Comment by rnugent3** [comment]
Looks like I tried to jump the gun! 

Good points. It seems like your criteria suggest a one or the other situation? 

In terms of maintenance and user experience, any thoughts on which wins?

**Comment by Brennan1994** [comment]
I have the feeling the both products could meet the needs of our design document. Admittedly, I haven't really looked into it, but I'd be shocked if that weren't true. They're both going to be able to view the relationships we want. so I don't want to spend too much time listing out the finer details there, when I'm pretty sure it'll be a wash. Feel free to correct me if you have a counter example in your pocket. That might be hubris. 

Proposing other Criteria
 - Have Existing Code
   - Having the library already used is a step up. Changing libraries is likely going to mean undoing previous work. 
 - Have Documentation
   - Does the library have resources and examples online for using it? 
 - Have In-House experience using the library
    - Do we have experts at HEC that could help us use the code? 
 - Is Actively developed and supported 
   - is this product actively maintained?
 - Is Free and Easy to Access
   - is this product going to be something we can access through NuGet? Will we need a License? Will this prevent outside HEC help from participating in development?
- Will have negative contract implications
   - Assuming we've thrown out the idea of using two libraries, which I think is fair. Is this going to cause undue pain in our contracting?

Do those seem like fine criteria?

**Reply by Brennan1994** [reply]
I also don't want to carry both dependencies, but I think it's also worth considering that we're getting close to the end of timeline to release a final product (relatively). If dropping one of these dependencies is a decision we can push to FDA2.1, or to tag on the end if we have extra time, I'll dejectedly advocate for that.

**Comment by rnugent3** [comment]
I just learned a new word!! I would advocate for the same, although not as dejectedly. I understand Will’s concerns, particularly as it relates to UX. If the dependencies behave sufficiently similar, and the implementations work, I’d like to leave it to 2.1.

**Comment by rnugent3** [comment]
If I understand correctly, this discussion has been resolved. I believe we're using OxyPlot with all the table with plot controls, and we're using SciChart for the IAS editor, and that we plan to swap out SciChart for a later version.
